### PR TITLE
[WIP] Receptor API client

### PR DESCRIPTION
**Pull request by slemrmartin**
Receptor client for communication with Receptor API/Kafka response.

---

* Receptor API: https://github.com/RedHatInsights/platform-receptor-controller

### CODE REVIEW

File: lib/topological_inventory/ansible_tower/receptor/response_worker.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+module TopologicalInventory::AnsibleTower
+  module Receptor
+    class ResponseWorker
+      def self.instance
+        @instance ||= self.new
```
Comment (agrare) [inline lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: Eh not a huge fan of the global singleton, could you just keep an instance in the connection and pass it wherever else is needed?

Comment (slemrmartin) [reply lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: ok, will do

Comment (agrare) [reply lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: :+1: thanks

Comment (slemrmartin) [reply lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: @agrare I refactored this singleton, but there is no place for instance variable. 
In this case singleton has reasonable usage, any other option doesn't improve stability or readability. So what exactly is the problem you have with singleton here?

File: lib/topological_inventory/ansible_tower/receptor/response_worker.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+      def start
+        return if started
+
+        self.started = true
+        Thread.new { listen }
+      end
+
+      def register_msg_id(msg_id, api_object, response_method = :response_received)
+        @registered_messages[msg_id] = { :api_object => api_object, :method => response_method }
```
Comment (agrare) [inline lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: This isn't guaranteed to be thread safe if you're using this to register from one thread while reading from it [here](https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/46/files#diff-2fde49ec8f16e8d8a0038a1e85c8dae3R44)

You could use `Concurrent::Map`

Comment (slemrmartin) [reply lib/topological_inventory/ansible_tower/receptor/response_worker.rb]: good point

**Comment by slemrmartin** [comment]
Replaced byt #71, closing

**Pull request closed without merging**